backend/common/db/migrations/versions/2025_12_19_0000-add_notifications_table.py

```
@@ -0,0 +1,112 @@
+        ('idx_notifications_user_created', ['user_id', 'created_at']),
+    ]
+    
+    for index_name, columns in indexes_to_create:
+        # Check if index exists
+        result = conn.execute(
+            sa.text(f"""
+                SELECT EXISTS (
+                    SELECT 1 FROM pg_indexes 
+                    WHERE indexname = '{index_name}'
+                )
+            """)
```

Ruff (0.14.10)

Possible SQL injection vector through string-based query construction

(S608)